test/WebJobs.Extensions.Tests/app.config

```diff
@@ -2,6 +2,7 @@
 <configuration>
   <appSettings>
     <add key="NotificationHubsConnectionString" value="ConnectionString_FromAppSettings" />
+    <add key="CosmosDBConnectionString" value="AccountEndpoint=https://someEndpoint;AccountKey=someKey;" />
```

Comment:
You can add a test that uses the `DefaultNameResolver` as the name resolver. Mocking the `INameResolver` lets you make sure all of the fallbacks work without needing to change environment variables a bunch. I think for the most part, we trust that the `DefaultNameResolver` does what we expect (I see a lot of usage of `TestNameResolver`s in our tests) and try to ensure that we're using the `INameResolver` interface correctly.